docs/guides/using-consumer-credential-resource.md

```diff
@@ -0,0 +1,200 @@
+Content-Length: 0
+Connection: keep-alive
+Server: gunicorn/19.9.0
+Date: Wed, 17 Jul 2019 19:25:32 GMT
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Credentials: true
+X-Kong-Upstream-Latency: 2
+X-Kong-Proxy-Latency: 1
+Via: kong/1.2.1
+```
+
+## Add authentication to the service
+
+With Kong, adding an authentication in front of an API is as simple as
+enabling a plugin.
+
+Let's add a KongPlugin resource to protect the API.
+
+```bash
+$ echo "
+apiVersion: configuration.konghq.com/v1
+kind: KongPlugin
+metadata:
+  name: httpbin-auth
+plugin: key-auth
+" | kubectl apply -f -
+kongplugin.configuration.konghq.com/httpbin-auth created
+```
+
+Now, associate this plugin with the previous Ingress rule we created
+using the `plugins.konghq.com` annotation.
+
+```bash
+$ echo "
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: demo
+  annotations:
+    plugins.konghq.com: httpbin-auth
+spec:
+  rules:
+  - http:
+      paths:
+      - path: /foo
+        backend:
+          serviceName: httpbin
+          servicePort: 80
+" | kubectl apply -f -
+```
+
+Any request matching the proxying rules defined in the `demo` ingress will
+now require a valid API key:
+
+```bash
+$ curl -i $PROXY_IP/foo/status/200
+HTTP/1.1 401 Unauthorized
+Date: Wed, 17 Jul 2019 19:30:33 GMT
+Content-Type: application/json; charset=utf-8
+Connection: keep-alive
+WWW-Authenticate: Key realm="kong"
+Content-Length: 41
+Server: kong/1.2.1
+
+{"message":"No API key found in request"}
+```
+
+As you can see above, Kong returns back a `401 Unauthorized` because
+we didn't provide an API key.
+
+## Provision a consumer
+
+Let's create a KongConsumer resource:
+
+```bash
+$ echo "apiVersion: configuration.konghq.com/v1
+kind: KongConsumer
+metadata:
+  name: harry
+username: harry" | kubectl apply -f -
+kongconsumer.configuration.konghq.com/harry created
+```
+
+Now, let's provision an API-key associated with
+this consumer so that we can pass the authentication imposed by Kong:
+
+```bash
+echo "apiVersion: configuration.konghq.com/v1
+kind: KongCredential
+metadata:
+  name: harry-apikey
+consumerRef: harry
+type: key-auth
+config:
+  key: sooper-secret-key" | kubectl apply -f -
+kongcredential.configuration.konghq.com/harry-apikey created
+```
+
+Here, we specify the tyep of credential as `key-auth` and then specify
+the API key using the `key` configuration value. This lets us declaratively
+provision credentials as well.
+
+## Use the credential
+
+Now, use the credential to pass authentication:
+
+```bash
+$ curl -i -H 'apikey: sooper-secret-key' $PROXY_IP/foo/status/200
+HTTP/1.1 200 OK
+Content-Type: text/html; charset=utf-8
+Content-Length: 0
+Connection: keep-alive
+Server: gunicorn/19.9.0
+Date: Wed, 17 Jul 2019 19:34:44 GMT
+Access-Control-Allow-Origin: *
+Access-Control-Allow-Credentials: true
+X-Kong-Upstream-Latency: 3
+X-Kong-Proxy-Latency: 1
+Via: kong/1.2.1
+```
+
+In this guide, we learnt how to leverage an authentication plugin in Kong
```

Comment:
```suggestion
In this guide, we learned how to leverage an authentication plugin in Kong
```

Reply:
learnt is most commonly used in British English, not so much in American English

Reply:
TIL (https://www.grammarly.com/blog/learned-learnt/). I grew up reading British English in school. Thank you for point this out!